localcache.go

```diff
@@ -270,11 +272,11 @@ func (c *Cache) Purge(older time.Duration) error {
 				return fmt.Errorf("invalid cache entry %q: %w", entry, err)
 			}
 			fileTime := time.Unix(0, ts)
-			if time.Since(fileTime) < older {
-				return nil
+			if clock.Since(fileTime) < older {
+				continue
```

Comment:
This was the actual bug.

Reply:
Error message with updated test:

    localcache_test.go:102: 
        	Error Trace:	localcache_test.go:102
        	Error:      	Should be empty, but was /var/folders/4r/nk2hgvs50352tpb02w4xhdqm0000gp/T/2139335582/2c/2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824
        	Test:       	TestPurge